### PR TITLE
Bump DSE version to 6.8.31 and OSS C* 3.x to 3.11.14

### PR DESCRIPTION
**Pull request by msmygit**
**What this PR does**: Bump OSS C* 3.x version to `3.11.14` and DSE version to `6.8.31`. There is a separate PR to update the `main` branch here at https://github.com/stargate/stargate/pull/2357.

**Which issue(s) this PR fixes**: n/a
Fixes #<issue number>: n/a

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by jeffreyscarpenter** [comment]
@msmygit the command is `/gcbrun` and it has to be run by a core contributor.

File: testing/pom.xml
Comment (tatu-at-datastax) [inline testing/pom.xml:380]: Where did this come from?

**Review by jeffreyscarpenter — APPROVED** [review]

**Comment by tatu-at-datastax** [comment]
This is good to get in as DSE-6.8.31 has Netty upgrades which will resolve at least 2 Snyk alerts.

File: persistence-cassandra-3.11/pom.xml
Comment (ivansenic) [inline persistence-cassandra-3.11/pom.xml:19]: As I stated in the other PR, this is wrong update:

image

File: persistence-cassandra-4.0/pom.xml
Comment (ivansenic) [inline persistence-cassandra-4.0/pom.xml:19]: And this one as well:

image

@tatu-at-datastax this change must bring a red flag when reviewing.. We already bumped the 4.x to `4.0.7` so either that PR was wrong or this one.. Please, let's not mechanically approve stuff, this has serious implications imo..

File: testing/pom.xml
Comment (ivansenic) [reply testing/pom.xml:380]: I also don't think we should make this change, I guess it was on purpose to have more than one node there.. Not sure..

**Review by ivansenic — COMMENTED** [review]
@msmygit @tatu-at-datastax I did not see this on time, however, I think there are changes here that are invalid. See below and check please..

File: persistence-cassandra-4.0/pom.xml
Comment (tatu-at-datastax) [reply persistence-cassandra-4.0/pom.xml:19]: @ivansenic I did not approve this PR, fwtw, and was suspicious of driver version change. But did not add a strong enough note to wait before merging.

**Comment by tatu-at-datastax** [comment]
This is also missing changes to `persistence-dse-6.8/pom.xml`... I think

**Pull request merged**
